client/my-sites/checkout/src/components/refund-policies.tsx

```diff
@@ -228,9 +233,7 @@ export function getRefundWindows( refundPolicies: RefundPolicy[] ): RefundWindow
 		}
 	} );
 
-	return Array.from( new Set( refundWindows ) ).filter(
-		( refundWindow ): refundWindow is RefundWindow => refundWindow !== undefined
```

Comment:
TIL about ["is" typeguard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)

Reply:
Yeah, what this type guard was saying was "we know the variable can be a `RefundWindow` or `undefined`. So if it's not `undefined`, then we will assume it's a `RefundWindow`. Which is correct, but mostly unnecessary because TypeScript can figure that out on its own. We mostly only need explicit `is` type guards when it's not clear what the object is and we need to perform some tests to be sure. For example, if a variable's type is `unknown` or its data comes from an API endpoint, we don't really know what's in there, and `is` tells TypeScript, "ok, we have tested this data and we are sure it's actually this specific type".